# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoadHandler.cpp
# Commit: 46d45c7bae2f
# Full Hash: 46d45c7bae2f0bce676b6103bb1c874c13bdcc11
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - Split ScriptLoadRequest into ScriptLoadRequest and DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134041
# ==============================================================================

diff -r a59c38e062bd -r 46d45c7bae2f dom/script/ScriptLoadHandler.cpp
--- a/dom/script/ScriptLoadHandler.cpp	Fri Feb 18 11:41:41 2022 +0000
+++ b/dom/script/ScriptLoadHandler.cpp	Fri Feb 18 12:45:48 2022 +0000
@@ -122,7 +122,7 @@
 
   if (!mPreloadStartNotified) {
     mPreloadStartNotified = true;
-    mRequest->NotifyStart(channelRequest);
+    mRequest->mLoadContext->NotifyStart(channelRequest);
   }
 
   if (mRequest->IsCanceled()) {
@@ -222,8 +222,8 @@
   // Check the hint charset from the script element or preload
   // request.
   nsAutoString hintCharset;
-  if (!mRequest->IsPreload()) {
-    mRequest->GetScriptElement()->GetScriptCharset(hintCharset);
+  if (!mRequest->mLoadContext->IsPreload()) {
+    mRequest->mLoadContext->GetScriptElement()->GetScriptCharset(hintCharset);
   } else {
     nsTArray<ScriptLoader::PreloadInfo>::index_type i =
         mScriptLoader->mPreloads.IndexOf(
@@ -296,7 +296,8 @@
 
   if (mRequest->IsLoadingSource()) {
     mRequest->SetTextSource();
-    TRACE_FOR_TEST(mRequest->GetScriptElement(), "scriptloader_load_source");
+    TRACE_FOR_TEST(mRequest->mLoadContext->GetScriptElement(),
+                   "scriptloader_load_source");
     return NS_OK;
   }
 
@@ -306,7 +307,7 @@
     cic->GetAlternativeDataType(altDataType);
     if (altDataType.Equals(nsContentUtils::JSBytecodeMimeType())) {
       mRequest->SetBytecode();
-      TRACE_FOR_TEST(mRequest->GetScriptElement(),
+      TRACE_FOR_TEST(mRequest->mLoadContext->GetScriptElement(),
                      "scriptloader_load_bytecode");
       return NS_OK;
     }
@@ -314,7 +315,8 @@
   }
 
   mRequest->SetTextSource();
-  TRACE_FOR_TEST(mRequest->GetScriptElement(), "scriptloader_load_source");
+  TRACE_FOR_TEST(mRequest->mLoadContext->GetScriptElement(),
+                 "scriptloader_load_source");
 
   MOZ_ASSERT(!mRequest->IsUnknownDataType());
   MOZ_ASSERT(mRequest->IsLoading());
@@ -339,11 +341,11 @@
 
   if (!mPreloadStartNotified) {
     mPreloadStartNotified = true;
-    mRequest->NotifyStart(channelRequest);
+    mRequest->mLoadContext->NotifyStart(channelRequest);
   }
 
-  auto notifyStop =
-      MakeScopeExit([&] { mRequest->NotifyStop(channelRequest, rv); });
+  auto notifyStop = MakeScopeExit(
+      [&] { mRequest->mLoadContext->NotifyStop(channelRequest, rv); });
 
   if (!mRequest->IsCanceled()) {
     if (mRequest->IsUnknownDataType()) {